modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -3570,6 +3570,8 @@ results are returned in the structure cv::Moments.
 
 @param array Raster image (single-channel, 8-bit or floating-point 2D array) or an array (
 \f$1 \times N\f$ or \f$N \times 1\f$ ) of 2D points (Point or Point2f ).
+Note: For Python users, note that the numpy type for the input array should be either np.int32 or
```

Comment:
I agree with @terfendail , it is better to use doxygen `@note` command to create separate section. It should be located after all `@param` and `@returns` sections.
The comment itself should say that it is only applicable to contour moments calculation, and that `np.float` will not work in this case because it is same as `np.float64` and doesn't satisy `Point2f` type check.